ktor-client/ktor-client-apache/jvm/src/io/ktor/client/engine/apache/ApacheHttpRequest.kt

```diff
@@ -36,13 +36,16 @@ internal suspend fun CloseableHttpAsyncClient.sendRequest(
         val statusLine = rawResponse.statusLine
 
         val status = HttpStatusCode(statusLine.statusCode, statusLine.reasonPhrase)
-        val version = with(rawResponse.protocolVersion) { HttpProtocolVersion.fromValue(protocol, major, minor) }
-        val headers = Headers.build {
-            rawResponse.allHeaders.forEach { headerLine ->
-                append(headerLine.name, headerLine.value)
-            }
+        val version = with(rawResponse.protocolVersion) {
+            HttpProtocolVersion.fromValue(protocol, major, minor)
         }
 
+        val rawHeaders = rawResponse.allHeaders.groupBy(
+            { it.name ?: "" },
```

Comment:
❓ I wonder what is an empty header name and what semantics does it have? Some clients use `null` or `""` as a special value for the status line.